docs/design/specs/Ecma-335-Augments.md

```diff
@@ -535,6 +545,13 @@ Interfaces may define static virtual methods that get resolved at runtime based
 
 ### II.12.2 Implementing virtual methods on interfaces
 
+(Add a new bullet under the second bullet in the bulleted list below the first paragraph by mentioning
+that classes can implement abstract static virtual methods by implementing a derived interface providing
+the default implementation for the abstract static virtual method):
```

Comment:
>abstract static virtual method [](http://example.com/codeflow?start=35&length=30)

This doesn't apply to non-abstract virtual methods?

Reply:
That is an interesting question. In another part of the spec augment (the bulleted list you commented on), I kept the pre-existing formulation *Interface methods are allowed to MethodImpl other interface methods the interface _requires_ (but we require the `MethodImpl`s to be final to keep things simple) - i.e. an interface is allowed to provide (or override) an implementation of another interface's method
  (instance or static virtual)*. I presume that the reference to `MethodImpl`s being *final* is supposed to mean that the SVM cannot be overridden multiple times at different places in the type hierarchy; this would likely also mean that a non-abstract static virtual method (i.e. one with implementation supplied at the point of declaration) cannot be overridden anywhere basically defeating the purpose of the `virtual` keyword. I don't see the need for this requirement and especially with default interface implementations of SVMs it no longer make much sense as we must traverse the interface hierarchies in well-defined order to achieve consistent results. I have updated the spec to this more general model; we'll see what @davidwrighton has to say on the subject.